gobcore/model/__init__.py

```diff
@@ -132,62 +156,90 @@ def get_inverse_relations(self):
         catalog = self.get_catalog(catalog_name)
         return catalog['collections'].keys() if 'collections' in catalog else None
 
     def get_collections(self, catalog_name):
+        """Deprecated.
+
+        Use:
+        * gob_model[catalog_name]['collections']
+        * if gob_model.get(catalog_name):
+              return gob_model[catalog_name].get('collections')
+        """
+        deprecation("deprecated: use gob_model[catalog_name]['collections'] or …")
         catalog = self.get_catalog(catalog_name)
         return catalog['collections'] if catalog and 'collections' in catalog else None
 
     def get_collection(self, catalog_name, collection_name):
+        """Deprecated.
+
+        Use:
+          try:
+              return gob_model[catalog_name]['collections'][collection_name]
+          except KeyError:
+              return None
+        """
+        deprecation("deprecated: use gob_model[catalog_name]['collections'].get(collection_name) …")
         collections = self.get_collections(catalog_name)
         return collections[collection_name] if collections and collection_name in collections else None
 
     def get_collection_by_name(self, collection_name):
-        """Finds collection only by name. Raises GOBException when multiple collections with name are found. Returns
-        (catalog, collection) tuple when success.
+        """Finds collection only by name.
+
+        Raises GOBException when multiple collections with collection_name are found.
+        Returns (catalog, collection) tuple when success.
 
         :param collection_name:
         :return:
         """
         collections = []
         catalog = None
 
-        for catalog_name in self._data.keys():
-            collection = self.get_collection(catalog_name, collection_name)
+        for catalog_name, catalog in self.items():
+            collection = catalog['collections'].get(collection_name)
 
             if collection:
                 collections.append(collection)
-                catalog = catalog_name
+                collection_catalog_name = catalog_name
 
         if len(collections) > 1:
             raise GOBException(f"Multiple collections found with name {collection_name}")
 
-        return (catalog, collections[0]) if collections else None
+        return (collection_catalog_name, collections[0]) if collections else None
 
     def has_states(self, catalog_name, collection_name):
-        """
-        Tells if a collection has states
+        """Tells if a collection has states.
 
         :param catalog_name: name of the catalog
         :param collection_name: name of the collection
         :return: True if the collection has states
         """
-        collection = self.get_collection(catalog_name, collection_name)
-        return collection.get("has_states") is True
+        try:
+            collection = self[catalog_name]['collections'].get(collection_name)
```

Comment:
`self[catalog_name]['collections'][collection_name]` seems more consistent with that KeyError exception handler already in place

Reply:
Agreed